File: scripts/R/lr_gender_sales.r
# Load required libraries
library(ggplot2)
library(dplyr)

# Assume columns are in order: Sales, GenderDummy
colnames(dataset) <- c('Sales', 'GenderDummy')

# Prepare the data
X <- dataset$GenderDummy
y <- dataset$Sales

# Create and fit the model
model <- lm(Sales ~ GenderDummy, data = dataset)

# Create a DataFrame with the results
results <- data.frame(
  Variable = c('Intercept', 'GenderDummy'),
  Coefficient = coef(model)
)

# Create a scatter plot
ggplot(dataset, aes(x = factor(GenderDummy), y = Sales)) +
  geom_jitter(color = '#0D6ABF', width = 0.2, alpha = 0.6) +
  labs(title = 'Sales vs GenderDummy', x = 'Gender', y = 'Sales') +
  scale_x_discrete(labels = c('0', '1')) +
  theme_minimal()

# Display the plot
print(last_plot())

# Display the results table
print(results)

# Return the results DataFrame for display in Power BI
results

File: scripts/Python/lr_gender_sales.py
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
import matplotlib.pyplot as plt

# Assume columns are in order: Sales, Income, Age, GenderDummy
dataset.columns = ['Sales', 'GenderDummy']

# Prepare the data
X = dataset[['GenderDummy']]
y = dataset['Sales']

# Create and fit the model
model = LinearRegression()
model.fit(X, y)

# Create a DataFrame with the results
results = pd.DataFrame({
    'Variable': ['Intercept', 'GenderDummy'],
    'Coefficient': [model.intercept_] + list(model.coef_)
})

# Create a scatter plot
plt.figure(figsize=(10, 6))
plt.scatter(dataset['Sales'], dataset['GenderDummy'], color='#0D6ABF')
plt.title('Sales vs GenderDummy')
plt.xlabel('Gender')
plt.ylabel('Sales')
plt.xticks([0, 1], ['0', '1'])
plt.xlim(-0.1, 1.1)
plt.show()

# Display the results table
print(results)

# Return the results DataFrame for display in Power BI
results